src/consensus/blockchain.py

```diff
@@ -515,6 +508,81 @@ async def pre_validate_blocks_multiprocessing(
         self,
         blocks: List[FullBlock],
     ) -> Optional[List[PreValidationResult]]:
-        return await pre_validate_blocks_multiprocessing(
-            self.constants, self.constants_json, self.sub_blocks, self.sub_height_to_hash, blocks, self.pool
+        return await pre_validate_blocks_multiprocessing(self.constants, self.constants_json, self, blocks, self.pool)
+
+    def sub_block_record(self, header_hash: bytes32) -> SubBlockRecord:
+        return self.__sub_blocks[header_hash]
+
+    def height_to_sub_block_record(self, sub_height: uint32) -> SubBlockRecord:
+        header_hash = self.sub_height_to_hash(sub_height)
+        return self.sub_block_record(header_hash)
+
+    def get_ses_heights(self) -> List[uint32]:
+        return sorted(self.__sub_epoch_summaries.keys())
+
+    def get_ses(self, height: uint32) -> SubEpochSummary:
+        return self.__sub_epoch_summaries[height]
+
+    def sub_height_to_hash(self, height: uint32) -> Optional[bytes32]:
+        return self.__sub_height_to_hash[height]
+
+    def contains_sub_height(self, height: uint32) -> bool:
+        return height in self.__sub_height_to_hash
+
+    def get_peak_height(self) -> Optional[uint32]:
+        return self.peak_height
+
+    async def warmup(self, fork_point: uint32):
+        if self.peak_height is None:
+            return
+        blocks = await self.block_store.get_sub_block_in_range(
```

Comment:
In the case that we are syncing from a deep reorg, this will load the whole blockchain basically, potentially millions of blocks. did you mean to load up to the fork point instead?